### PR TITLE
emit error message when Java backend compiler get confused

### PR DESCRIPTION
**Pull request by yilongli**
@andreistefanescu @dwightguth please review. BTW, I would love to have better error message if possible. @andreistefanescu Any suggestions? Thanks.

### CODE REVIEW

File: src/javasources/KTool/src/org/kframework/compile/utils/CompileDataStructures.java [hunk 1]
```diff
@@ -106,15 +106,7 @@ public ASTNode visit(KApp node, Void _)  {
...
-        // TODO(AndreiS): the lines below should work one KLabelConstant are properly created
-        //if (kLabelConstant.productions().size() != 1) {
-        //    /* ignore KLabels associated with multiple productions */
-        //    return super.transform(node);
-        //}
-        //Production production = kLabelConstant.productions().iterator().next();
-
-        if (context.productionsOf(kLabelConstant.getLabel()).size() != 1) {
-            /* ignore KLabels associated with multiple productions */
+        if (context.productionsOf(kLabelConstant.getLabel()).isEmpty()) {
```
Comment (dwightguth) [inline src/javasources/KTool/src/org/kframework/compile/utils/CompileDataStructures.java]: Are you sure isEmpty() and size() != 1 are equivalent here?

Comment (dwightguth) [inline src/javasources/KTool/src/org/kframework/compile/utils/CompileDataStructures.java]: oh wait never mind, I see it now

File: src/javasources/KTool/src/org/kframework/compile/utils/CompileDataStructures.java [hunk 2]
```diff
@@ -123,6 +115,19 @@ public ASTNode visit(KApp node, Void _)  {
         if (sort == null) {
             return super.visit(node, _);
         }
+        
+        // TODO(AndreiS): the lines below should work one KLabelConstant are properly created
+        if (context.productionsOf(kLabelConstant.getLabel()).size() > 1) {
```
Comment (dwightguth) [inline src/javasources/KTool/src/org/kframework/compile/utils/CompileDataStructures.java]: should this be moved up?

Comment (yilongli) [inline src/javasources/KTool/src/org/kframework/compile/utils/CompileDataStructures.java]: I need it to first pass the above check to make sure `production.getSort()` is actually a data structure sort.

Comment (dwightguth) [inline src/javasources/KTool/src/org/kframework/compile/utils/CompileDataStructures.java]: oh okay

**Comment by dwightguth** [comment]
lgtm

**Comment by andreistefanescu** [comment]
lgtm

**Comment by yilongli** [comment]
Just pushed another minor refactoring. I will merge this when the tests pass.

**Pull request merged**